dependencies-check-java/action.yml

```diff
@@ -20,34 +20,47 @@ runs:
     - name: Get base POM
       shell: bash
       run: |
-        curl -H "Accept: application/vnd.github+json" -H "Authorization: token ${{ inputs.github_token }}" https://api.github.com/repos/${{ github.repository }}/contents/${{ inputs.root_folder }}/pom.xml | jq -r ".content" | base64 -di >> ${{ inputs.root_folder }}/base-pom.xml
+        curl -H "Accept: application/vnd.github+json" -H "Authorization: token ${{ inputs.github_token }}" https://api.github.com/repos/${{ github.repository }}/contents/${{ inputs.root_folder }}/pom.xml | jq -r ".content" > base64_file
+        grep "null" base64_file || status=$?
+        if [ $status -eq 0 ]; then
+          echo "BASE_POM_NOT_FOUND=1" >> $GITHUB_ENV
+          echo "Base POM not found"
+        else
+          echo "BASE_POM_NOT_FOUND=0" >> $GITHUB_ENV
+          cat base64_file | base64 -di > ${{ inputs.root_folder }}/base-pom.xml
+        fi
 
     - name: Save/Restore base dependencies from cache
       uses: actions/cache@v3
+      if: ${{ env.BASE_POM_NOT_FOUND == 0 }}
       with:
         key: ${{ runner.os }}-maven-${{ hashFiles('**/base-pom.xml') }}
         path: |
           ~/.m2
 
     - name: Get dependencies from base POM
       shell: bash
+      if: ${{ env.BASE_POM_NOT_FOUND == 0 }}
       run: |
-        mvn -f ${{ inputs.root_folder }}/base-pom.xml dependency:list -Dsort=true -DforceStdout -DexcludeTransitive=true -DoutputFile=base-deps.txt --no-transfer-progress
+        mvn -f ${{ inputs.root_folder }}/base-pom.xml dependency:list -Dsort=true -DforceStdout -DincludeScope=runtime -DoutputFile=base-deps.txt --no-transfer-progress
```

Comment:
Shouldn't -DexcludeScope=provided be added here to let dependencies provided with the core only be listed/checked in the core repository?

Reply:
According to the [documentation](https://maven.apache.org/plugins/maven-dependency-plugin/list-mojo.html#includeScope): 

> `runtime` include scope gives runtime and compile dependencies

While for excludeScope:

> Scope threshold to exclude, if no value is defined for include

Reply:
It says that runtime includes compile. If you look at compile documentation, it says 

> `compile` include scope gives compile, provided, and system dependencies,

So if provided gets included, I'm sure you will get a lot of unrelevant dependencies.

Reply:
But it could be that it is just a misinterpretation from my side, and that "compile dependencies" is not the same as "compile scope". Then the PR is ok

Reply:
For the record, the doc says "The scope threshold value being interpreted is the scope as Maven filters for creating a classpath, not as specified in the pom"